### PR TITLE
Add blog post for 1.58.1

### PR DESCRIPTION
**Pull request by pietroalbini**
There isn't really much we can write on these regressions other than a slightly longer description of each item in the release notes. Let me know if I should expand on some of them.

r? @Mark-Simulacrum 
cc @rust-lang/release

### CODE REVIEW

File: posts/2022-01-20-Rust-1.58.1.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+## What's in 1.58.1 stable
+
+Rust 1.58.1 addresses several regressions in Rust, Clippy and Rustfmt
+introduced in the 1.58.0 stable release. You can find more detailed information
+on the specific regressions in the [release notes].
+
+[release notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1581-2022-01-20
+
+### Contributors to 1.58.0
```
Comment (yerke) [inline posts/2022-01-20-Rust-1.58.1.md:32]:
```suggestion
### Contributors to 1.58.1
```

Comment (pietroalbini) [reply posts/2022-01-20-Rust-1.58.1.md]: Whelp, damn copy-paste. Fixed!

**Comment by pietroalbini** [comment]
Expanded the blog post with a description of the regressions.

**Comment by pietroalbini** [comment]
Added the note about CVE-2022-21658.

File: posts/2022-01-20-Rust-1.58.1.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+Rust 1.58.1 also addresse several regressions introduced in Rust 1.58.0:
+
+* The `non_send_fields_in_send_ty` Clippy lint was discovered to have too many
+  false positives and has been moved to the experimental lints group (called
+  "nursery").
+* The `useless_format` Clippy lint has been updated to handle captured
+  identifiers in format strings, introduced in Rust 1.58.0.
+* A regression in Rustfmt preventing generated files from being formatted when
+  passed through the standard input has been fixed.
+* An incorrect error message displayed by Rustc in some cases have been fixed.
```
Comment (Mark-Simulacrum) [inline posts/2022-01-20-Rust-1.58.1.md:41]:
```suggestion
* An incorrect error message displayed by `rustc` in some cases have been fixed.
```

**Review by Mark-Simulacrum — APPROVED** [review]
Looks good modulo some small nits.

File: posts/2022-01-20-Rust-1.58.1.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+Rust 1.58.1 also addresses several regressions in diagnostics and tooling introduced in Rust 1.58.0:
+
+* The `non_send_fields_in_send_ty` Clippy lint was discovered to have too many
+  false positives and has been moved to the experimental lints group (called
+  "nursery").
+* The `useless_format` Clippy lint has been updated to handle captured
+  identifiers in format strings, introduced in Rust 1.58.0.
+* A regression in Rustfmt preventing generated files from being formatted when
+  passed through the standard input has been fixed.
+* An incorrect error message displayed by `rustc` in some cases have been fixed.
```
Comment (pietroalbini) [inline posts/2022-01-20-Rust-1.58.1.md:41]:
```suggestion
* An incorrect error message displayed by `rustc` in some cases has been fixed.
```

**Pull request merged**